test/test.py

```diff
@@ -1388,6 +1394,100 @@ def test_branches(self):
             response = self.test_client.post(form['action'], data=data, follow_redirects=True)
             self.assertFalse('Not Allowed' in response.data)
 
+    def test_get_request_does_not_create_branch(self):
+        ''' Navigating to a made-up URL should not create a branch
+        '''
+        with HTTMock(self.mock_persona_verify):
+            self.test_client.post('/sign-in', data={'email': 'erica@example.com'})
+
+        with HTTMock(self.auth_csv_example_allowed):
+            fake_branch_name = 'this-should-not-create-a-branch'
+            #
+            # edit
+            #
+            response = self.test_client.get('/tree/{}/edit/'.format(fake_branch_name), follow_redirects=True)
+            self.assertEqual(response.status_code, 200)
+            # the branch path should not be in the returned HTML
+            self.assertFalse('/{}'.format(fake_branch_name) in response.data)
+            # the branch name should not be in git's branches list
+            self.assertFalse(fake_branch_name in self.origin.branches)
+
+            #
+            # history
+            #
+            response = self.test_client.get('/tree/{}/history/'.format(fake_branch_name), follow_redirects=True)
+            self.assertEqual(response.status_code, 200)
+            # the branch path should not be in the returned HTML
+            self.assertFalse('/{}'.format(fake_branch_name) in response.data)
+            # the branch name should not be in git's branches list
+            self.assertFalse(fake_branch_name in self.origin.branches)
+
+            #
+            # review
+            #
+            response = self.test_client.get('/tree/{}/review/'.format(fake_branch_name), follow_redirects=True)
+            self.assertEqual(response.status_code, 200)
+            # the branch path should not be in the returned HTML
+            self.assertFalse('/{}'.format(fake_branch_name) in response.data)
+            # the branch name should not be in git's branches list
+            self.assertFalse(fake_branch_name in self.origin.branches)
+
+            #
+            # view
+            #
+            response = self.test_client.get('/tree/{}/view/'.format(fake_branch_name), follow_redirects=True)
+            self.assertEqual(response.status_code, 200)
+            # the branch path should not be in the returned HTML
+            self.assertFalse('/{}'.format(fake_branch_name) in response.data)
+            # the branch name should not be in git's branches list
+            self.assertFalse(fake_branch_name in self.origin.branches)
```

Comment:
`self.origin.branches` is updated in the process kicked off by the http request at the start of each block.